src/Plugin/resource/Field/ResourceFieldEntity.php

```diff
@@ -782,15 +772,26 @@ protected function entityTypeWrapper() {
   /**
    * {@inheritdoc}
    */
-  public function getBundles() {
-    return $this->bundles;
+  public function getBundle() {
+    return $this->bundle;
   }
 
   /**
    * {@inheritdoc}
    */
-  public function setBundles($bundles) {
-    $this->bundles = $bundles;
+  public function setBundle($bundle) {
+    // Do not do pointless work if not needed.
+    if (!empty($this->bundle) && $this->bundle == $bundle) {
+      return;
+    }
+    $this->bundle = $bundle;
+
+    // If this is an options call, then introspect Entity API to add more data
+    // to the public field information.
+    if ($this->getRequest()->getMethod() != RequestInterface::METHOD_OPTIONS) {
```

Comment:
yeah, I had a big chunk of code before so I wanted to save an indentation level. The I refactored it into a function.